lib/sles4sap.pm

```diff
@@ -386,15 +387,27 @@ sub copy_media {
     my ($self, $proto, $path, $nettout, $target) = @_;
     my $mnt_path = '/mnt';
     my $media_path = "$mnt_path/" . get_required_var('ARCH');
-    my $jobtoken = get_required_var('JOBTOKEN');
+
+    # Set some NFS options in case we are using NFS
+    my $sha = Digest::SHA->new('sha256');
+    $sha->add(get_required_var('JOBTOKEN'));
+    my $nfs_client_id = $sha->hexdigest;
+    my $options = 'ro';
+    if ($proto eq 'nfs') {
+        my $nfs_timeo = get_var('NFS_TIMEO');
+        $options = $nfs_timeo ? "timeo=$nfs_timeo,rsize=16384,wsize=16384,ro" : 'rsize=16384,wsize=16384,ro';
+    }
 
     # First create $target and copy media there. A
     assert_script_run "mkdir $target";
-    assert_script_run "mount -t $proto -o ro $path $mnt_path";
+    assert_script_run "mount -t $proto -o $options $path $mnt_path", 90;
     # Attempt to force a unique NFSv4 client id
-    enter_cmd "echo $jobtoken | sha256sum | awk '{print \$1}' | tee /sys/fs/nfs/net/nfs_client/identifier";
+    assert_script_run 'while (! ls /sys/fs/nfs/net/nfs_client/identifier); do sleep 1; done';    # Wait for file to be available
+    assert_script_run 'chmod +w /sys/fs/nfs/net/nfs_client/identifier';    # Ensure file is writable
+    script_run "echo $nfs_client_id > /sys/fs/nfs/net/nfs_client/identifier";
+    script_run 'cat /sys/fs/nfs/net/nfs_client/identifier';
```

Comment:
I suggest to either check the return code or use `assert_script_run` to prevent hidden failures and long unnoticed timeouts. It's unlikely this fails but possible. Good practice is to always use assert_script_run unless you want to branch based on return code

Reply:
These 2 can actually be calls to `enter_cmd` instead of `script_run`, but we wanted to discard issues with the NFS client ID as set by this code, and `enter_cmd` was in some tests causing the output of the command called with `enter_cmd` and the one from the next line (L409) to appear combined in the results making it more difficult to debug. I decided to use `script_run` and not check the retcode on purpose, but will change these 2 to `enter_cmd` after we confirm this method works and is stable.